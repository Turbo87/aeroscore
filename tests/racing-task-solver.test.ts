import {generateRacingTest} from './generators';

generateRacingTest('2017-07-17-lev');
generateRacingTest('2017-07-17-lev', '2017-07-17T17:00:00Z');
generateRacingTest('Bayreuth/2018-05-30/Open');
