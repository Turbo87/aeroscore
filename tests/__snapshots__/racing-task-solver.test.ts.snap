// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RacingTaskSolver 2017-07-17-lev 1`] = `
"
   1	  IGP	  1.040	  14:50:57	  01:48:14	  136.1 km	   75.47 km/h	   431
   2	  1G 	  1.010	  14:54:34	  01:56:21	  136.1 km	   70.21 km/h	   398
   3	  P9 	  1.040	  14:51:22	  01:58:40	  136.1 km	   68.84 km/h	   362
   4	  MI 	  1.040	  14:51:15	  01:59:41	  136.1 km	   68.25 km/h	   356
   5	  999	  1.000	  14:51:52	  02:06:39	  136.1 km	   64.49 km/h	   344
   6	  LC 	  1.040	  14:53:46	  02:02:47	  136.1 km	   66.53 km/h	   338
   7	  MS 	  0.980	  14:52:13	  02:12:04	  136.1 km	   61.85 km/h	   329
   8	  E1 	  1.040	  14:50:58	  02:04:32	  136.1 km	   65.60 km/h	   329
   9	  1R 	  1.040	  14:52:58	  02:05:15	  136.1 km	   65.22 km/h	   325
  10	  SW 	  0.980	  14:52:26	  02:13:19	  136.1 km	   61.27 km/h	   323
  11	  1  	  1.010	  14:47:05	  02:09:50	  136.1 km	   62.91 km/h	   320
  12	  FR 	  0.980	  14:51:34	  02:14:25	  136.1 km	   60.77 km/h	   317
  13	  H1 	  1.050	  14:53:02	  02:06:09	  136.1 km	   64.75 km/h	   314
  14	  EI 	  1.070	  14:51:15	  02:04:12	  136.1 km	   65.77 km/h	   311
  15	  sq 	  1.070	  14:53:01	  02:08:54	  136.1 km	   63.37 km/h	   287
  16	  JM 	  1.010	  14:56:52	  02:17:36	  136.1 km	   59.36 km/h	   283
  17	  XN 	  0.980	  14:51:30	  02:23:14	  136.1 km	   57.03 km/h	   276
  18	  T1 	  1.040	  15:06:25	  02:18:02	  136.1 km	   59.18 km/h	   262
  19	  WJ 	  1.040	  15:06:22	  02:30:44	  136.1 km	   54.19 km/h	   211
  20	  33 	  0.980	  14:51:16	          	  103.6 km	             	   130
  21	  ZG 	  1.040	  14:06:55	          	   94.2 km	             	   111
"
`;

exports[`RacingTaskSolver 2017-07-17-lev until 2017-07-17T17:00:00Z 1`] = `
"
   1	  IGP	  1.040	  14:50:57	  01:48:14	  136.1 km	   75.47 km/h	   417
   2	  1G 	  1.010	  14:54:34	  01:56:21	  136.1 km	   70.21 km/h	   399
   3	  P9 	  1.040	  14:51:22	  01:58:40	  136.1 km	   68.84 km/h	   379
   4	  MI 	  1.040	  14:51:15	  01:59:41	  136.1 km	   68.25 km/h	   375
   5	  999	  1.000	  14:51:52	  02:06:39	  136.1 km	   64.49 km/h	   368
   6	  LC 	  1.040	  14:53:46	  02:02:47	  136.1 km	   66.53 km/h	   365
   7	  E1 	  1.040	  14:50:58	  02:04:32	  136.1 km	   65.60 km/h	   360
   8	  1R 	  1.040	  14:52:58	  02:05:15	  136.1 km	   65.22 km/h	   358
   9	  1  	  1.010	  14:47:05	  02:09:50	  136.1 km	   62.91 km/h	   355
  10	  H1 	  1.050	  14:53:02	  02:06:09	  136.1 km	   64.75 km/h	   351
  11	  EI 	  1.070	  14:51:15	  02:04:12	  136.1 km	   65.77 km/h	   350
! 12	  MS 	  0.980	  13:55:53	          	  126.3 km	             	   257
! 13	  FR 	  0.980	  14:46:35	          	  124.2 km	             	   253
! 14	  SW 	  0.980	  13:59:06	          	  124.1 km	             	   252
! 15	  sq 	  1.070	  14:00:07	          	  131.0 km	             	   244
! 16	  XN 	  0.980	  14:51:30	          	  114.4 km	             	   233
! 17	  JM 	  1.010	  14:56:52	          	  114.2 km	             	   225
! 18	  33 	  0.980	  14:51:16	          	  103.6 km	             	   211
! 19	  T1 	  1.040	  14:48:31	          	  109.7 km	             	   210
! 20	  WJ 	  1.040	  14:48:16	          	  100.5 km	             	   193
  21	  ZG 	  1.040	  14:06:55	          	   94.2 km	             	   181
"
`;

exports[`RacingTaskSolver Bayreuth/2018-05-30/Open 1`] = `
"
   1	  HS 	  1.000	  12:43:08	  03:03:56	  312.3 km	  101.87 km/h	  1000
   2	  FO 	  1.000	  12:42:38	  03:04:25	  312.3 km	  101.60 km/h	   995
   3	  VX 	  1.000	  12:42:51	  03:05:59	  312.3 km	  100.75 km/h	   978
   4	  FJ 	  1.000	  12:50:08	  03:06:16	  312.3 km	  100.60 km/h	   975
   5	  RB 	  1.000	  12:40:25	  03:06:44	  312.3 km	  100.35 km/h	   970
   6	  NY 	  1.000	  12:40:03	  03:06:57	  312.3 km	  100.22 km/h	   968
   7	  WR 	  1.000	  12:46:49	  03:10:41	  312.3 km	   98.27 km/h	   929
   8	  FU 	  1.000	  12:40:15	  03:10:45	  312.3 km	   98.23 km/h	   929
   9	  VY 	  1.000	  12:40:01	  03:10:47	  312.3 km	   98.22 km/h	   928
  10	  ET 	  1.000	  12:23:31	  03:11:47	  312.3 km	   97.70 km/h	   918
  11	  AM 	  1.000	  12:44:38	  03:12:19	  312.3 km	   97.43 km/h	   913
  12	  VH 	  1.000	  12:46:05	  03:13:20	  312.3 km	   96.92 km/h	   903
  13	  KV 	  1.000	  12:44:21	  03:13:44	  312.3 km	   96.72 km/h	   899
  14	  IM 	  1.000	  12:42:56	  03:13:53	  312.3 km	   96.65 km/h	   897
  15	  YY 	  1.000	  12:35:10	  03:16:52	  312.3 km	   95.18 km/h	   869
  16	  LY 	  1.000	  12:40:22	  03:18:29	  312.3 km	   94.40 km/h	   853
  17	  MM 	  1.000	  12:24:36	  03:19:57	  312.3 km	   93.71 km/h	   840
  18	  NX 	  1.000	  12:09:18	  03:26:42	  312.3 km	   90.65 km/h	   780
  19	  MD 	  1.000	  12:36:48	  03:26:50	  312.3 km	   90.59 km/h	   779
  20	  TEX	  1.000	  12:36:39	  03:27:02	  312.3 km	   90.51 km/h	   777
  21	  QAX	  1.000	  12:14:59	  03:40:51	  312.3 km	   84.84 km/h	   666
  22	  W1 	  1.000	  12:42:45	  03:44:11	  312.3 km	   83.58 km/h	   641
  23	  DF 	  1.000	  12:14:51	  03:44:15	  312.3 km	   83.56 km/h	   641
  24	  EPI	  1.000	  12:32:09	  03:53:53	  312.3 km	   80.11 km/h	   573
  25	  LM 	  1.000	  12:30:24	  03:55:54	  312.3 km	   79.43 km/h	   559
  26	  TOY	  1.000	  12:48:05	  04:13:40	  312.3 km	   73.87 km/h	   450
  27	  JOY	  1.000	  12:48:01	  04:24:07	  312.3 km	   70.94 km/h	   393
"
`;
